server/api.js

```diff
@@ -349,6 +349,83 @@ router.get("/buyer-tender", async (req, res) => {
 		: res.status(500).send({ code: "SERVER_ERROR" });
 });
 
+router.post("/tender/:tenderId/status", async (req, res) => {
+	const tenderId = parseInt(req.params.tenderId, 10);
+	const newStatus = req.body.status;
+	const validStatuses = ["Active", "In Review", "Closed", "Awarded"];
+	const user = req.user;
+
+	if (!validStatuses.includes(newStatus)) {
+		return res.status(400).send({ code: "INVALID_STATUS" });
+	}
+
+	let client;
+
+	try {
+		client = await pool.connect();
+
+		await client.query("BEGIN");
+
+		const tenderResult = await client.query(
+			"SELECT buyer_id, status FROM tender WHERE id = $1;",
+			[tenderId]
+		);
+
+		if (tenderResult.rowCount === 0) {
+			await client.query("ROLLBACK");
+			return res.status(404).send({ code: "TENDER_NOT_FOUND" });
+		}
+
+		const tender = tenderResult.rows[0];
+		const currentStatus = tender.status;
+		const buyerId = tender.buyer_id;
+
+		if (user.id !== buyerId) {
+			await client.query("ROLLBACK");
+			return res.status(403).send({ code: "FORBIDDEN" });
+		}
+
+		if (currentStatus === "Awarded" || currentStatus === "Closed") {
+			await client.query("ROLLBACK");
+			return res.status(400).send({ code: "NO_CHANGE_ALLOWED" });
+		}
+
+		if (currentStatus === "Active") {
+			if (newStatus !== "In Review" && newStatus !== "Closed") {
+				await client.query("ROLLBACK");
+				return res.status(400).send({ code: "INVALID_STATUS_TRANSITION" });
+			}
+		} else if (currentStatus === "In Review") {
+			if (newStatus !== "Closed") {
+				await client.query("ROLLBACK");
+				return res.status(400).send({ code: "INVALID_STATUS_TRANSITION" });
+			}
+		}
```

Comment:
What are the allowed transitions? It might be clearer to have a map of allowed transitions. Something like:

```
const allowedTenderStatusTransitions = {
   "In Review": ["Closed"],
   ...
}
```